core/src/main/scala/org/http4s/ContentCoding.scala

```diff
@@ -114,7 +114,11 @@ object ContentCoding {
   }
 
   implicit val http4sOrderForContentCoding: Order[ContentCoding] =
-    Order.by(c => (c.coding.toLowerCase, c.qValue))
+    Order.from {
+      case (a, b) if a.coding.compareToIgnoreCase(b.coding) == 0 => a.qValue.compare(b.qValue)
+      case (a, b) => a.coding.compareToIgnoreCase(b.coding)
+    }
```

Comment:
I think this is right (I haven't rerun the test with the seed), but you could avoid repeating the comparison when codings aren't equal.  I think this is also right:

```suggestion
    val cmp = a.coding.compareToIgnoreCase(b.coding)
    if (cmp == 0) a.qValue.compare(b.qValue) else cmp
```

Reply:
@rossabaker this does not work. The test fails with that seed value. Will change the PR to WIP and investigate.

Reply:
Nuts. Maybe I misanalyzed it on the ticket.  If we can figure out the exact inputs to that failing case, and not just the `.toString` representation, it should be much easier to spot that error.

Reply:
I opened a PR after investigating deeper myself, but after that, I think this might still be correct.  We just don't have an example where it matters, because characters are restricted to token characters, none of which do weird things in case folding.